crawler.py

```diff
@@ -87,15 +89,26 @@ def test_crash(driver):
 var crash = badptr.contents;""")
 
 
-def get_domain_list(n_sites):
+def get_domain_list(n_sites, exclude_option):
     """Get the top n sites from the tranco list"""
-    tr = Tranco(cache=False)
-
-    if not n_sites:
-        n_sites = DEFAULT_NUM_SITES
-
-    return tr.list().top(n_sites)
-
+    tr = Tranco(cache=False).list().top()
+    extract = TLDExtract(cache_file=False)
+    domains = []
+
+    # if the exclude TLD option is passed in, remove those TLDs
+    if exclude_option:
+        excluded_tlds = exclude_option.split(",")
+        # check for first occurring domains in list that don't have excluded TLD
+        for domain in tr:
+            if extract(domain).suffix not in excluded_tlds:
+                domains.append(domain)
+            # return list of acceptable domains if it's the correct length
+            if len(domains) == n_sites:
+                return domains
+    # if no exclude option is passed in, just return top n domains from list
+    else:
+        domains = tr[0 : n_sites-1]
```

Comment:
Why the -1?

Reply:
it's to make sure the zero-indexed domain gets counted toward the `n_sites` goal, otherwise the length of the list would be `n_sites`+1

Reply:
```
$ python
>>> l = ['one', 'two', 'three']
>>> n = 2
>>> print(len(l[:n-1]))
1
>>> l[:n-1]
['one']
>>> l[:n]
['one', 'two']
```

Reply:
Oh wow, I was thinking of the javascript convention that *is* inclusive of the final index given when slicing an array. Thanks for pointing that out!

Reply:
It's less important to memorize this kind of stuff than to check your work.

How do you know the code you just wrote does what you want it to do without running it and testing the significant scenarios, that is the different branches your new code takes.

What happens when the new setting wasn't provided? What happens when it was but set to something that will end up excluding almost all domains? ...

Is the output correct? Was the expected number of domains visited?

You can simplify testing by limiting to the total data set you're testing against (change `top()` to `top(20)` in `get_domain_list()`).

Reply:
You're right, I should have tested that particular part more carefully. Chopping the last index off seemed obvious to me; I was definitely confusing it with a different convention. Thanks for catching it!